client/src/routes/config.ts

```diff
@@ -136,4 +141,25 @@ export class MainRouterConfig extends FoundationRouterConfiguration<LoginSetting
       },
     });
   }
+
+  private async reAuthFromSession(settings: LoginSettings, phase: NavigationPhase) {
+    return this.auth.reAuthFromSession().then((authenticated) => {
+      logger.info(`reAuthFromSession. authenticated: ${authenticated}`);
+      if (authenticated) {
+        this.redirectIfNotPermitted(settings, phase);
+      }
+      return authenticated;
+    });
+  }
+
+  private redirectIfNotPermitted(settings: LoginSettings, phase: NavigationPhase) {
+    const { path } = phase.route.endpoint;
+    if (settings?.isPermitted && !settings.isPermitted()) {
+      const route = 'not-permitted';
```

Comment:
If we make a typo in the route here, it'll just silently fail. We can assign this route/path to a variable and reuse e.g. not-permitted route component could provide an export with default route.

Reply:
[f2b3bd6](https://github.com/genesiscommunitysuccess/blank-app-seed/pull/255/commits/f2b3bd671bdca43dc2469035ec93c92559367164)